### DISCUSSION TITLE
[Discussion/Proposal] OData "prefixes" should be replaced with MVC Areas

### DISCUSSION THREAD

**Discussion by julealgon** [OData/AspNetCoreOData#455] [Ideas]
The "prefix" idea in OData is extremely old, and always served as a means to have multiple EDM models in the same project. The prefix effectively provides a means to isolate one "API" from another. The whole thing is completely custom, non-standard AspNet MVC.

I'd like to understand if it would be possible to just leverage MVC Areas instead, and get rid of the custom aspects of OData prefixes. There would still be a way to tell the framework that a different EDM model was to be used, but this would be then keyed by the MVC area.

Areas serve basically the same purpose as OData prefixes from my understanding: isolating parts of an API in a different "scope", by adding a prefix to the route. Similarly to how v8 leverages standard MVC routing, I think OData should also leverage standard MVC area functionality instead of the whole new "OData prefix" concept. This would make the API more seamless to use and more consistent.

From what I can tell, OData doesn't deal well with Areas at all today, and it would become a huge mess if _both_ OData prefixes _and_ areas were to be supported at the same time.

Related:
- https://github.com/OData/AspNetCoreOData/issues/77

**Comment by xuzhg** [comment]
@julealgon I enabled discussion. Is it ok to convert this issue to discussion?

**Comment by julealgon** [comment]
> @julealgon I enabled discussion. Is it ok to convert this issue to discussion?

Of course, go ahead @xuzhg . 

Good idea enabling discussions... I checked before creating the issue and would've used that instead if it was available.
